scripts/validate.js

```diff
@@ -6,51 +6,71 @@ const { join } = require('path');
 
 const { listDir } = require('hexo-fs');
 const { log } = hexo;
-const sizeOf = require('image-size');
+const sharp = require('sharp');
 
-async function validateThemeNames() {
+function difference(setA, setB) {
+  const diff = new Set(setA);
+  for (const elem of setB) {
+    diff.delete(elem);
+  }
+  return diff;
+}
+
+async function validateTheme() {
   let isValidationPassed = true;
+
   const themeData = hexo.locals.get('data').themes;
-  const themeNames = [];
-  let name = '';
+  const themes = new Set();
+  const duplicate = new Set();
   for (const theme of themeData) {
-    name = String(theme.name).toLocaleLowerCase();
-    if (themeNames.indexOf(name) !== -1) {
-      isValidationPassed = false;
-      break;
+    const name = theme.name.toLocaleLowerCase();
+    if (themes.has(name)) {
+      duplicate.add(name);
     } else {
-      themeNames.push(name);
+      themes.add(name);
```

Comment:
Set will be deduped automatically, no need to wrap it in `else`. But it is fine.